### ISSUE TITLE
NVCC errors while compiling the boost headers

### ISSUE THREAD

**Issue by Jark5455** [boostorg/beast#2644]
Hello, I created a basic application to use boost beast and boost asio with CUDA. However, it appears that nvcc runs into a lot of errors while compiling the headers. The full error output is [here](https://paste.gg/p/anonymous/51c42314259c408e95992de93a8ec6f9). 

By any chance are there any precompiled headers for boost that I could use instead?

**Comment by klemens-morgenstern** [comment]
Are you including a `detail` header?

**Comment by Jark5455** [comment]
I don’t believe so, the only headers I am using are `boost/beast.hpp` and `boost/asio.hpp`
